### PR TITLE
Add description on how we handle security issues in Airflow

### PR DESCRIPTION
**Pull request by potiuk**
This is a proposed description of the security policy we have in Apache Airflow.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by pierrejeambrun — APPROVED** [review]

File: .github/SECURITY.rst [hunk 1]
```diff
@@ -52,3 +52,75 @@ before disclosing it publicly.
...
+  (including their employers) unless specifically authorised by the security team members, specifically
+  if diagnosing and solving the issue might involve the need of external experts - for example security
+  experts that are available through Airflow stakeholders. The intent about involving 3rd parties has
+  to be discussed and agreed up at security@airflow.apache.org.
+
+* They have to have an `ICLA <https://www.apache.org/licenses/contributor-agreements.html>`_ signed with
+  Apache Software Foundation.
+
+* The security team members might inform 3rd parties about fixes, for example in order to asses if the fix
+  is solving the problem or in order to assess it's applicability to be applied by 3rd parties, as soon
```
Comment (pierrejeambrun) [inline .github/SECURITY.rst]:
```suggestion
  is solving the problem or in order to assess its applicability to be applied by 3rd parties, as soon
```

File: .github/SECURITY.rst [hunk 2]
```diff
@@ -52,3 +52,75 @@ before disclosing it publicly.
...
+  as PR solving the issue is opened in the public airflow repository.
```
Comment (pankajkoti) [inline .github/SECURITY.rst]:
```suggestion
  as a PR solving the issue is opened in the public airflow repository.
```

File: .github/SECURITY.rst [hunk 3]
```diff
@@ -52,3 +52,75 @@ before disclosing it publicly.
...
+
+Releasing Airflow with security patches
+---------------------------------------
+
+Apache Airflow uses strict `SemVer <https://semver.org>`_ versioning policy, which means that we strive for
+any release of a given ``MAJOR`` Version (version "2" currently) to be backwards compatible. When we
+release ``MINOR`` version, the development continues in the ``main`` branch where we prepare the next
+``MINOR`` version, but we release ``PATCHLEVEL`` releases with selected bugfixes (including security
+bugfixes) cherry-picked to the latest released ``MINOR`` line of Apache Airflow. At the moment we
+release a new ``MINOR`` version we stop releasing ``PATCHLEVEL`` releases for the previous ``MINOR`` version.
```
Comment (pankajkoti) [inline .github/SECURITY.rst]:
```suggestion
bugfixes) cherry-picked to the latest released ``MINOR`` line of Apache Airflow. At the moment, when we
release a new ``MINOR`` version, we stop releasing ``PATCHLEVEL`` releases for the previous ``MINOR`` version.
```

File: dev/README_RELEASE_HELM_CHART.md [hunk 4]
```diff
@@ -715,6 +730,7 @@ Don't forget to thank the folks who tested and close the issue tracking the test
 
 Updating issue templates in `.github/ISSUE_TEMPLATE/airflow_helmchart_bug_report.yml` with the new version
 
+
```
Comment (pankajkoti) [inline dev/README_RELEASE_HELM_CHART.md]:
```suggestion
```

**Review by potiuk — COMMENTED** [review]
Apply Review comments

**Review by pankajkoti — APPROVED** [review]
Unqualified to review yet, but would still like to give a +1 as it looks great to me.

**Comment by potiuk** [comment]
> Unqualified to review yet, but would still like to give a +1 as it looks great to me.

That's still draft - initially discussed in the PMC. we are going to get it to the devlist soon.

**Review by uranusjr — APPROVED** [review]

File: .github/SECURITY.rst [hunk 5]
```diff
@@ -52,3 +52,69 @@ before disclosing it publicly.
...
+* They are not supposed to reveal the information about pending and unfixed security issues to anyone
+  (including their employers) unless specifically authorised by the security team members, specifically
+  if diagnosing and solving the issue might involve the need of external experts - for example security
+  experts that are available through Airflow stakeholders. The intent about involving 3rd parties has
+  to be discussed and agreed up at security@airflow.apache.org.
+
+* They have to have an `ICLA <https://www.apache.org/licenses/contributor-agreements.html>`_ signed with
+  Apache Software Foundation.
+
+* The security team members might inform 3rd parties about fixes, for example in order to asses if the fix
```
Comment (vincbeck) [inline .github/SECURITY.rst]:
```suggestion
* The security team members might inform 3rd parties about fixes, for example in order to assess if the fix
```

Comment (potiuk) [reply .github/SECURITY.rst]: Right :).

**Pull request merged**